torchdp/per_sample_gradient_clip.py

```diff
@@ -125,39 +126,58 @@ def calc_thresh_value(self, layer_norms):
             thresh_norm if self.clip_per_layer else thresh_norm * len(self.named_params)
         )
 
-    def clip(self) -> List[float]:
+    def clip(self, virtual: bool = False) -> (List[float], int):
```

Comment:
nit: change bool arg name to `is_virtual`?

Reply:
Fixed